cpu/sam0_common/include/periph_cpu_common.h

```diff
@@ -767,6 +768,48 @@ typedef struct {
     uint32_t muxpos;       /**< ADC channel pin multiplexer value */
 } adc_conf_chan_t;
 
+/**
+ * @name Ethernet peripheral parameters
+ * @{
+ */
+#ifndef ETH_RX_BUFFER_COUNT
+#define ETH_RX_BUFFER_COUNT (4)
+#endif
+
+#ifndef ETH_TX_BUFFER_COUNT
+#define ETH_TX_BUFFER_COUNT (4)
+#endif
+
+#ifndef ETH_RX_BUFFER_SIZE
+#define ETH_RX_BUFFER_SIZE (1536)
+#endif
+
+#ifndef ETH_TX_BUFFER_SIZE
+#define ETH_TX_BUFFER_SIZE (1536)
```

Comment:
Is this needed? E.g. on the STM32 the iolist is directly fed into the DMA, as the DMA supports gather scatter. One copy less should speed up the TX path.

Also: If gather-scatter is also supported for RX, maybe you could use smaller buffers? If those in total are at least big enough for the MTU, there driver can still handle huge frames. But I believe that many small frames are more likely for embedded systems.

Reply:
The GMAC IP and this driver should be able to manage multiple small buffers. (RX and TX descriptor have start and end bits for the frame so we can parse the buffer descriptor to know how many buffers contain the whole frame). I was thinking to do so but as I couldn't figure out which buffer size would be the best or how many buffers I should use, I decided to go the easy way. a few buffer with the maximum frame size I found in the datasheet for GMAC IP.

I can try to modify this PR to follow this behavior if you want.

Reply:
Maybe it is better to keep it simple and do the optimization as follow up :-)

Reply:
> But I believe that many small frames are more likely for embedded systems.

IIRC some industrial / real time IP cameras use jumbo frames to transmit image data.
But I agree that we should keep things simple for now.

Reply:
I think the STM32 Ethernet driver could cope both, as size and number of RX frames are configurable. Once this drivers gains (as follow up) support to also split received frames over multiple small buffers, I think this is due it its lower RAM requirements a reasonable default. And people willing to trade in RAM for better throughput can still do so.

Just after writing that I remember reading that the STM32 Ethernet block needs a flag set to no longer discard frames bigger than 1500 Bytes. I guess we should expose that as netopt. And maybe the default value of that setting could be based on the RX buffer size, as there hardly is a reason to provide more than 1500 Bytes per RX buffer without having jumbo frames in mind.